jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -536,23 +536,23 @@ static ecma_object_t *
 ecma_op_find_super_declerative_lex_env (ecma_object_t *lex_env_p) /**< starting lexical enviroment */
 {
   JERRY_ASSERT (lex_env_p);
-  JERRY_ASSERT (ecma_op_resolve_super_reference_value (lex_env_p));
   JERRY_ASSERT (ecma_get_lex_env_type (lex_env_p) != ECMA_LEXICAL_ENVIRONMENT_SUPER_OBJECT_BOUND);
 
-  while (true)
+  while (lex_env_p != NULL)
   {
     ecma_object_t *lex_env_outer_p = ecma_get_lex_env_outer_reference (lex_env_p);
 
-    JERRY_ASSERT (lex_env_outer_p);
```

Comment:
Are you sure this is the right fix? When can it be `NULL`?

Reply:
If you check the reported issue, the `eval` call in the `Animal` constructor is on the top of the inheritance chain, however, this class has has been created first and has no super, so it cannot have superDeclarativeLexEnv.

Reply:
Thanks for the explanation.